src/routes/room.js

```diff
@@ -1,17 +1,36 @@
 const router = require('express').Router();
 const StudyRoom = require('../models/studyRoom.model');
 const Student = require('../models/student.model');
-const CourseNotes = require('../models/courseNotes.model');
-const { createStudyRoom, editStudyRoom } = require("../factory/roomFactory");
-const createCourseNotes = require("../factory/courseNotesFactory");
+const { createStudyRoom } = require("../factory/roomFactory");
+const CourseNotesRepository = require("../repository/courseNotesRepository")
+const StudyRoomRepository = require('../repository/studyRoomRepository');
+const StudentRepository = require('../repository/studentRepository');
+const studyRoomRepository = require('../repository/studyRoomRepository');
 
 
 router.route('/').put((req, res) => {
-  StudyRoom.findOne({ studyRoomID: req.body.studyRoomID.toString() })
-    .then(() => {
-      const room = editStudyRoom(req.body)
-      room.save()
-        .then(() => res.json(`Room ${room.studyRoomID} updated`))
+  StudyRoomRepository.findOne(req.body.studyRoomID)
```

Comment:
Add `const data = req.body;` here. Variable `data` is not defined, so if conditions aren't executing properly.